pkg/zfs/zfs_util.go

```diff
@@ -364,12 +364,20 @@ func CreateClone(vol *apis.ZFSVolume) error {
 	if vol.Spec.FsType == "xfs" {
 		// for mounting the cloned volume for xfs, a new UUID has to be generated
 		device := ZFS_DEVPATH + volume
-		cmd := exec.Command("xfs_admin", "-U", "generate", device)
+		cmd := exec.Command("xfs_repair", "-L", device)
```

Comment:
Q_: while taking snapshot on a xfs mounted volume, is there something that can be done to flush the logs?

Reply:
there are no options available as such, There is xfs_freeze option there but that is also not helping. So followed the solution which there are following for LVM snapshot :-
https://learn.redhat.com/t5/Platform-Linux/How-to-Mounting-an-XFS-LVM-snapshot/td-p/287

Reply:
there is another solution which I have suggested on the bug(https://github.com/openebs/zfs-localpv/issues/56) was use below storageclass for xfs cloned volumes :-
```
kind: StorageClass
metadata:
  name: openebs-zfspv1
allowVolumeExpansion: true
parameters:
  fstype: "xfs"
  poolname: "zfspv-pool"
provisioner: zfs.csi.openebs.io
volumeBindingMode: WaitForFirstConsumer
mountOptions:
  - nouuid
```
As with nouuid it will not check for the UUID to mount the fs.
```
       nouuid Don't check for double mounted file systems using the file
              system uuid.  This is useful to mount LVM snapshot volumes,
              and often used in combination with "norecovery" for mounting
              read-only snapshots.
```

Reply:
makes sense. one additional comment. Can we also include a comment in the code here as to why we are doing this?

Reply:
After reading so much about this, I am really not sure what is the right fix for it. The only thing which worried me with the current fix is this line "Note that destroying the log may cause corruption -- please attempt a mount of the filesystem before doing this." Now I am thinking, should we ask user to use nouuid mount option for this in the storageclass for the cloned volumes, as with that it will replay the log.
```
mountOptions:
  - nouuid
```

Reply:
thanks for working with the XFS community and coming with this solution of:
- temporary mount with no-uuid, 
- after the logs are replayed, generate new uuid and mount it.